File: Data_Structure_Review/Queue_Implementation/queue.py
from collections import deque

# in Python we normally use 双向队列类 dequeue as queue to use;
# Even though the queue.Queue() is the pure queue class, but not recommended;

# Init
que: deque[int] = deque()

# Add elt goes into the queue
que.append(1)
que.append(3)
que.append(2)
que.append(5)
que.append(4)

# Accessing the front of the queue
head: int = que[0]

# elt goes out from the queue, normally also happens at the front
pop: int = que.popleft()

# Get the size of this queue
size: int = len(que)

# emptyness check
is_empty: bool = len(que) == 0

File: Data_Structure_Review/Queue_Implementation/Queue.java
package Queue_Implementation;

import java.util.ArrayDeque;
import java.util.Deque;

public class Queue {
    public static void main(String[] args){

        // Init
        Deque<Integer> que = new ArrayDeque<>();

        // Push elt into the queue; append at the tail;
        que.addLast(1);
        que.addLast(3);
        que.addLast(2);
        que.addLast(5);
        que.addLast(4);

        // Accessing the first elt in the queue; 
        int head = que.peekFirst(); // safe peek; return null when empty; 
        head = que.getFirst(); // use when expect non-empty; and wants an error when empty --- Throws NoSuchElementException

        // first elt in the queue goes out from queue --- pop operation
        int pop = que.removeFirst(); // use when expect non-empty; and wants an error when empty --- Throws NoSuchElementException
        pop = que.pollFirst(); // safe pop; return null when empty; 

        // Get the size of the queue
        int size = que.size();

        // Emptyness check
        Boolean isEmpty = que.isEmpty();

        System.out.println("head = " + head);
        System.out.println("pop  = " + pop);
        System.out.println("size = " + size);
        System.out.println("isEmpty = " + isEmpty);
        System.out.println("queue = " + que); // prints remaining elements in order
    }
}
